encoding/ssz/query/testutil/runner.go

```diff
@@ -0,0 +1,40 @@
+package testutil
+
+import (
+	"testing"
+
+	"github.com/OffchainLabs/prysm/v6/encoding/ssz/query"
+	"github.com/OffchainLabs/prysm/v6/testing/assert"
+	ssz "github.com/ferranbt/fastssz"
+)
+
+func RunStructTest(t *testing.T, spec TestSpec) {
+	t.Run(spec.Name, func(t *testing.T) {
+		info, err := query.AnalyzeSSZInfo(spec.Type)
+		assert.NoError(t, err)
+
+		testInstance := spec.Instance
+		marshaller, ok := testInstance.(ssz.Marshaler)
+		assert.Equal(t, true, ok, "Test instance must implement ssz.Marshaler, got %T", testInstance)
+
+		marshalledData, err := marshaller.MarshalSSZ()
+		assert.NoError(t, err)
+
+		for _, pathTest := range spec.PathTests {
+			t.Run(pathTest.Path, func(t *testing.T) {
+				path, err := query.ParsePath(pathTest.Path)
+				assert.NoError(t, err)
+
+				_, offset, length, err := query.CalculateOffsetAndLength(info, path)
+				assert.NoError(t, err)
+
+				expectedRawBytes := marshalledData[offset : offset+length]
+				assert.Equal(t, uint64(len(expectedRawBytes)), length, "Extracted value length mismatch: got %d, want %d", len(expectedRawBytes), length)
```

Comment:
I don't think this is very useful. It can only fail if the offset is too big (in which case there won't be enough bytes in `marshalledData` to go from `offset` to `offset+length`), and in such case the last assert will fail anyway.